tests/fault_management_test.go

```diff
@@ -0,0 +1,369 @@
+// Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"context"
+	"os"
+	"testing"
+	"tests/helpers"
+	"time"
+
+	"github.com/nvidia/nvsentinel/commons/pkg/statemanager"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+func TestDryRunMode(t *testing.T) {
+	feature := features.New("TestDryRunMode").
+		WithLabel("suite", "fault-quarantine-special-modes")
+
+	var testCtx *helpers.QuarantineTestContext
+	var originalDeployment *appsv1.Deployment
+	var podNames []string
+	testNamespace := "immediate-test"
+
+	feature.Setup(func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		client, err := c.NewClient()
+		require.NoError(t, err)
+
+		dryRunEnabled := true
+		var newCtx context.Context
+		newCtx, testCtx, originalDeployment = helpers.SetupQuarantineTestWithOptions(ctx, t, c,
+			"data/basic-matching-configmap.yaml",
+			&helpers.QuarantineSetupOptions{
+				DryRun: &dryRunEnabled,
+			})
+
+		// Create test namespace and pods to verify full pipeline is blocked
```

Comment:
Can we remove the obvious comments in net-new code? If the code is structured in a way it is easily understandable, we won't need to have many comments